front/components/agent_builder/transformAgentConfiguration.ts

```diff
@@ -0,0 +1,225 @@
+  const formActions: AgentBuilderAction[] = [];
+
+  // Transform MCP server configurations to form actions
+  for (const action of actions) {
+    if (!isServerSideMCPServerConfiguration(action)) {
+      continue;
+    }
+
+    // Find matching action type using our mapping
+    const actionType = Object.entries(MCP_SERVER_TO_ACTION_TYPE_MAP).find(
+      ([serverName]) =>
+        isInternalMCPServerOfName(
+          action.internalMCPServerId,
+          serverName as InternalMCPServerNameType
+        )
+    )?.[1];
+
+    if (actionType) {
+      formActions.push(transformMCPServerConfigToAction(action, actionType));
+    }
+  }
+
+  // Add data visualization action if enabled (this is stored as a flag, not an MCP config)
+  if (visualizationEnabled) {
+    formActions.push({
+      id: "data_visualization",
+      name: "Data Visualization",
+      description: "Enable data visualization capabilities",
+      type: "DATA_VISUALIZATION",
+      noConfigurationRequired: true,
+      configuration: {
+        type: "DATA_VISUALIZATION",
+      },
+    });
+  }
+
+  return formActions;
+}
+
+function transformMCPServerConfigToAction(
+  mcpConfig: ServerSideMCPServerConfigurationType,
+  actionType: SupportedAgentBuilderActionType
+): AgentBuilderAction {
+  const dataSourceConfigurations = transformDataSourceConfigurations(
+    mcpConfig.dataSources || []
+  );
+
+  const baseActionData = {
+    id: mcpConfig.sId,
+    name: mcpConfig.name,
+    description: mcpConfig.description || "",
+    noConfigurationRequired: false,
+  };
+
+  if (actionType === "EXTRACT_DATA") {
+    return {
+      ...baseActionData,
+      type: "EXTRACT_DATA",
+      configuration: {
+        type: "EXTRACT_DATA",
+        dataSourceConfigurations,
+        timeFrame: mcpConfig.timeFrame,
+        jsonSchema: mcpConfig.jsonSchema,
+      },
+    };
+  } else if (actionType === "INCLUDE_DATA") {
+    return {
+      ...baseActionData,
+      type: "INCLUDE_DATA",
+      configuration: {
+        type: "INCLUDE_DATA",
+        dataSourceConfigurations,
+        timeFrame: mcpConfig.timeFrame,
+      },
+    };
+  } else {
+    return {
+      ...baseActionData,
+      type: "SEARCH",
+      configuration: {
+        type: "SEARCH",
+        dataSourceConfigurations,
+      },
+    };
+  }
+}
+
+function transformDataSourceConfigurations(
+  dataSources: DataSourceConfiguration[]
+): DataSourceViewSelectionConfigurations {
+  const configurations: DataSourceViewSelectionConfigurations = {};
+
+  for (const ds of dataSources) {
+    configurations[ds.dataSourceViewId] = {
+      dataSourceView: {
+        sId: ds.dataSourceViewId,
+      } as DataSourceViewType,
+      selectedResources:
+        (ds.filter?.parents?.in?.map((internalId: string) => ({
+          internalId,
+        })) as DataSourceViewContentNode[]) || [],
+      isSelectAll: !ds.filter?.parents,
+      tagsFilter: ds.filter?.tags || null,
+    };
+  }
+
+  return configurations;
+}
```

Comment:
Need to rework this, not quite sure what's the most elegant approach but this is definitely not good enough